### DISCUSSION TITLE
How to get additional information from a signed pdf file

### DISCUSSION THREAD

**Discussion by areqq** [MatthiasValvekens/pyHanko#185] [Q&A] [answered]
I have pdf signed on https://autenti.com/ (enclosed)
They added page with signatures:
![image](https://user-images.githubusercontent.com/15180289/204031580-ab795728-485a-4066-b2e0-ad32c5c42048.png)

I'm looking for idea how to get info who signed.
in binary pdf file I see:
`/Filter/Adobe.PPKLite/Location()/M(D:20221125172749+01'00')/Prop_Build<</App<</Name/Autenti#20sp.#20z#20o.o.#20z#20siedzib#05#20w#20Poznaniu#20#2861-806#29#20przy#20ul.#20Zwi#19ty#20Marcin#2029#2f8.>>>>/Reason(Podpisanie dokumentu;Franek Kimono;franek.kimono@mailnesia.com;5260211104)/SubFilter/ETSI.CAdES.detached/Type/Sig>>`

How to get this using pyHanko?

[Test_document.pdf](https://github.com/MatthiasValvekens/pyHanko/files/10094443/Test_document.pdf)

**Comment by MatthiasValvekens** [comment, marked as answer]
Hi @areqq,

There's an `embedded_signatures` property on `PdfFileReader` that allows you to loop over the signatures in the document. The API for `EmbeddedPdfSignature` is [available in the documentation](https://pyhanko.readthedocs.io/en/latest/api-docs/pyhanko.sign.validation.pdf_embedded.html#pyhanko.sign.validation.pdf_embedded.EmbeddedPdfSignature). Not all of the properties you mention are exposed directly, but you can get access to the signature dictionary itself via the `sig_object` field. In other words, to get the **Reason** field, you'd write something like

```
emb_sig = r.embedded_signatures[0]
reason = emb_sig.sig_object['/Reason']
```
